packet_buffer.go

```diff
@@ -11,15 +11,17 @@ import (
 // packetBuffer represents a packet buffer
 type packetBuffer struct {
 	packetSize       int
+	filter           PacketFilter
 	r                io.Reader
 	packetReadBuffer []byte
 }
 
 // newPacketBuffer creates a new packet buffer
-func newPacketBuffer(r io.Reader, packetSize int) (pb *packetBuffer, err error) {
+func newPacketBuffer(r io.Reader, packetSize int, filter PacketFilter) (pb *packetBuffer, err error) {
```

Comment:
Could you rename this variable from `filter` to `s` ?

Reply:
Done